server/src/main/java/io/whitefox/api/deltasharing/loader/DeltaShareTableLoader.java

```diff
@@ -0,0 +1,24 @@
+package io.whitefox.api.deltasharing.loader;
+
+import io.whitefox.persistence.memory.PTable;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+@ApplicationScoped
+public class DeltaShareTableLoader implements TableLoader<DeltaSharedTable> {
```

Comment:
This does not load the table, it caches it, at least I would change the name ;)
Also, tables change over time, I’m not sure caching it is a good idea.